interfaces/builtin/pulseaudio.go

```diff
@@ -20,30 +20,170 @@
 package builtin
 
 import (
+	"bytes"
+
 	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
 )
 
-const pulseaudioConnectedPlugAppArmor = `
+var pulseaudioConnectedPlugAppArmor = []byte(`
+capability dac_override,
+
+/etc/machine-id r,
+
+/{run,dev}/shm/pulse-shm-* rwk,
+
+# Running as system instance
+owner /{,var/}run/pulse/ r,
+owner /{,var/}run/pulse/native rwk,
+`)
+
+var pulseaudioConnectedPlugAppArmorDesktop = []byte(`
+# Only on desktop do we need access to /etc/pulse for any PulseAudio client
+# to read available client side configuration settings. On an Ubuntu Core
+# device those things will be stored inside the snap directory.
 /etc/pulse/ r,
 /etc/pulse/* r,
```

Comment:
In addition to these desktop-only accesses, shouldn't we also add these for the pulse cookie (from Touch policy):
owner @{HOME}/.pulse-cookie        rk,
owner @{HOME}/.config/pulse/cookie rk,

and when are these needed (if ever):
owner @{HOME}/.pulse/              r,
owner @{HOME}/.pulse/*             rk,